nri-prelude/src/Task.hs

```diff
@@ -222,6 +224,15 @@ background task =
         Prelude.pure <| Ok ()
     )
 
+backgroundPool :: Int -> Task Never a -> Task x ()
+backgroundPool poolSize task =
+  Internal.Task
+    ( \handler ->
+        Async.Pool.withTaskGroup (Prelude.fromIntegral poolSize) <| \g -> do
```

Comment:
If I'm reading thing correctly, the goal is to run this on one of `n` task pools, lazily creating them if they aren't there yet.

So, `backgroundPool 1 someTask` would add that task to "pool 1", `backgroundPool 2 someTask` would add that task to "pool 2", where the assumption is that each "pool" would roughly correspond to a thread, and the goal is to prevent resource starvation.

Assuming that understanding is correct:

it looks like `withTaskGroup` creates a new pool on every invocation and then adds a taskgroup with the specified size to that pool. So that's an unbounded growth on the number of pools, only limiting the amount of works that can happen in parallel within each taskgroup.

So I don't think those two match 🤔 Correct me if I'm wrong tho!

@jali-clarke might have some idea for how to divvy up work over a predefined number of workers, tho! At the very least, your input is welcome here 😄

Reply:
I'm going to fix this up properly tomorrow. On my phone now and with kids :-)

Reply:
actually, maybe I'm just going to do this in place where it's needed, because I'm not sure what's a nice API yet.

Reply:
Additionally, we've used `Data.Pool` in other places so I'm going to do that as well.